src/routes.rs

```diff
@@ -83,9 +92,56 @@ pub async fn root_html() -> Result<Response, Rejection> {
     <html>
         <head>
             <title>Root</title>
+            <meta name="viewport" content="width=device-width, initial-scale=1">
+        </head>
+        <body>
+            <h1>Session Open Group Server</h1>
+            <p>This is a Session open group server.</p>
+        </body>
+    </html>
+    "#;
+    return Ok(warp::reply::html(body).into_response());
+}
+
+pub async fn fallback_html(room: String, query_map: HashMap<String, String>) -> Result<Response, Rejection> {
+    let public_key = match query_map.get("public_key") {
+        Some(val) => val,
+        None => "",
+    };
+    if public_key == "" || room == "" {
```

Comment:
Is `public_key` even important here? I don't see it used anywhere below.

Reply:
I've chose to do this because if someone opens http(s)://HOST/ANYTHING without the public_key they get the generic 500 error.  
This is because the instructions in the page tell you to copy-paste the URL into Session, but if the URL doesn't have the public key, then it will not work.  
This is kind of half way between zero checks (not even checking if the public_key is there), and checking if the room ID and name are correct, which, as I mentioned, is a potential security concern.  
It's really a design choice, and I'm not in charge of making those, so if you prefer to do the full checks or do none, just tell me, and I'll gladly update the code.

Reply:
@nielsandriesse let us know :P

Reply:
Sorry, missed the @ before. I think I agree with @msgmaxim here that it'd be nice to show some message telling the user that they forgot to include the public key in this scenario.